include/picongpu/particles/collision/InterCollision.hpp

```diff
@@ -29,480 +29,425 @@
+                = srcCollisionFunctor(worker, localSuperCellOffset, densityLongList, densityShortList, maxListLength);
 
-                    return destCollisionFunctor;
-                }
-            };
+            if constexpr(useScreeningLength)
+            {
+                auto const shifted = screeningLengthSquared.shift(superCellIdx * SuperCellSize::toRT());
+                auto const idxInSuperCell = pmacc::math::mapToND(SuperCellSize::toRT(), linearCellIdx);
+                destCollisionFunctor.coulombLogFunctor.screeningLengthSquared_m = shifted(idxInSuperCell)[0];
+            }
 
-            /* Run kernel for collisions between two species.
-             *
-             * @tparam T_CollisionFunctor A binary particle functor defining a single macro particle collision in
-             * the binary-collision algorithm.
-             * @tparam T_FilterPair A pair of particle filters, each for each species
-             *     in the colliding pair.
-             * @tparam T_Species0 1st colliding species.
-             * @tparam T_Species1 2nd colliding species.
-             */
-            template<
-                typename T_CollisionFunctor,
-                typename T_FilterPair,
-                typename T_Species0,
-                typename T_Species1,
-                uint32_t colliderId,
-                uint32_t pairId>
-            struct DoInterCollision
+            if(minListLength != 0u)
             {
-                /* Run kernel
-                 *
-                 * @param deviceHeap A pointer to device heap for allocating particle lists.
-                 * @param currentStep The current simulation step.
-                 */
-                HINLINE void operator()(const std::shared_ptr<DeviceHeap>& deviceHeap, uint32_t currentStep)
+                uint32_t const numDupParticles = maxListLength - minListLength;
+                for(uint32_t i = 0; i < numDupParticles; ++i)
                 {
-                    using Species0 = T_Species0;
-                    using FrameType0 = typename Species0::FrameType;
-                    using Filter0 = typename T_FilterPair::first ::template apply<Species0>::type;
+                    auto par0 = parAccessor0[i % size0];
+                    auto par1 = parAccessor1[i % size1];
+                    destCollisionFunctor.duplicationCorrection
+                        = duplicationCorrection(i, minListLength, maxListLength);
+                    destCollisionFunctor(detail::makeCollisionContext(worker, rngHandle), par0, par1);
+                }
+                destCollisionFunctor.duplicationCorrection = 1u;
+                for(uint32_t i = numDupParticles; i < maxListLength; ++i)
+                {
+                    auto par0 = parAccessor0[i % size0];
+                    auto par1 = parAccessor1[i % size1];
+                    destCollisionFunctor(detail::makeCollisionContext(worker, rngHandle), par0, par1);
+                }
```

Comment:
```suggestion
                for(uint32_t i = numDupParticles; i < maxListLength; ++i)
                {
                    destCollisionFunctor.duplicationCorrection
                        = duplicationCorrection(i, minListLength, maxListLength);
                    auto par0 = parAccessor0[i % size0];
                    auto par1 = parAccessor1[i % size1];
                    destCollisionFunctor(detail::makeCollisionContext(worker, rngHandle), par0, par1);
                }
```

This was missing. Also there is no need to set it to 1, we are changing it anyway.